app/tests/algorithms_tests/test_views.py

```diff
@@ -1357,11 +1359,34 @@ def test_create_job_is_idempotent(
         assert Job.objects.count() == 1
 
     @override_settings(task_eager_propagates=True, task_always_eager=True)
-    def test_create_job_with_faulty_file_input(
+    @pytest.mark.django_db
+    @pytest.mark.parametrize(
+        "file_json_content,job_status,error_message,detailed_error_message,ci_count,upload_exists",
+        [
+            (
+                b'{"Foo": "bar"}',
+                Job.CANCELLED,
+                "One or more of the inputs failed validation.",
+                {
+                    "ci_json_file": "JSON does not fulfill schema: instance is not of type 'array'"
+                },
+                0,
+                True,
+            ),
+            (b'[{"Foo": "bar"}]', Job.PROVISIONED, "", {}, 1, False),
+        ],
+    )
+    def test_create_job_with_file_input(
         self,
```

Comment:
I think this test should only test the faulty file input - that's what the test name suggests. So no parametrization necessary. There should be another test for the correct file input, one that comes to mind is `test_create_job_with_multiple_new_inputs` - this test exists for the API create view and the UI view.

Reply:
Done.
I retracted the parametrisation and added a check for upload file deletion in `test_create_job_with_multiple_new_inputs` for both the view and API.